### ISSUE TITLE
 [example/imagenet_classification] Continue training of pre-trained Inception-BN as prep step for finetuning

### ISSUE THREAD

**Issue by revilokeb** [apache/incubator-mxnet#1439]
As exercise for fine-tuning I would like to continue training the inception-bn pretrained network on a subset of the imagenet 2012 training data using a new net with identical architecture to inception-bn (possibly later modifying learning rates in a subsequent step). Based on the example/imagenet_classification I have done the following steps:
1. Load the pre-trained model: loaded_model = mx.model.FeedForward.load(prefix = Inception-BN, load_epoch = 39, ctx = mx.gpu())
2. Load the network structure for the new net: import importlib, network = importlib.import_module('symbol_incpetion-bn').get_symbol(1000)
3. Store the model params of the pre-trained model: model_args = {'arg_params' : loaded_model.arg_params, 'aux_params': loaded_model.aux_params, 'begin_epoch' : load_epoch}
4. Create new model with arg_params and aux_params of the loaded model: model_new = mx.model.FeedForward( ctx = [gpu(0), gpu(1)], symbol = network, num_epoch = 41, learning_rate = 0.001, initializer = mx.init.Xavier(factor_type = "in", magnitude = 2.34, **model_args) 
5. Fit the model, here TrainIterator and ValIterator both contain the same subset from imagenet 2012 training data: model_new.fit(X = TrainIterator, eval_data = ValIterator, batch_end_callback = mx.callback.Speedometer(batch_size = 128, 50), eval_batch_end_callback = mx.callback.Speedometer(128, 50))

Contrary to my expectation of starting with good training and validation accuracy when doing the above train accuracy starts at 0.001, so around random.

On the other hand when simply continuing training the loaded model by doing model.fit(...) in step 5. instead of model_new.fit(...) I am starting at accuracy around 0.65, validation accuracy however always stays always around 0.001 in both approaches. 

I have manually checked my DataIterator to have correct labels (aligned to imagenet convention), for both TrainIterator and ValIterator, labels are also correct and in line with what I get when predicting from pre-trained inception-bn net.

Any hints what I am missing? Any input appreciated, thanks and best!

**Comment by piiswrong** [comment]
Why do you need step 2 to 5? You can directly train the model created with step 1.

**Comment by revilokeb** [comment]
Ok thanks a lot for your feedback!

What then can I do if I want to modify, say, the number of neurons in last layer e.g. for fine-tuning? I thought the way to go in mxnet is to 1. create a new network with some modified layers, e.g. different number of neurons in the last layer, 2. transfer the weights / bias from pre-trained network to new network -> this should give identical weights / bias for pre-trained and new network except for last layer, 3. start fine-tuning the new network (after setting learning rates appropriately, with blockGrad or set_lr_scale)

Should be done differently?

**Comment by piiswrong** [comment]
You can use mx.init.Load

**Comment by revilokeb** [comment]
When continue learning pre-trained inception-bn model from model gallery on whole imagenet 1000 class training data (prepared own .rec-file using im2rec, manually checked for correct labeling) I would expect (training) accuracy at around what is cited on https://github.com/dmlc/mxnet-model-gallery/blob/master/imagenet-1k-inception-bn.md

However when I do the following, I was surprised to find learning to start from scratch (i.e. at training accuracy 0.001): 
python example/image-classification/train-imagenet.py --load-epoch=39 --model-prefix=PATH_TO_PRETRAINED_MODEL/Inception_BN --data-dir=PATH_TO_IMAGENET_ITERATORS --train-dataset=imagenet_train.rec --val-dataset=imagenet_val.rec

It seems that the pre-trained model is somehow re-initialized with random weights / not appropriately loaded from Inception_BN-0039.params. Can train-imagenet.py not be used to continue training pre-trained model from model gallery?

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
